tests/unit/s2n_tls13_new_session_ticket_test.c

```diff
@@ -0,0 +1,141 @@
+                    ticket_key.data, ticket_key.size, current_time/ONE_SEC_IN_NANOS));
+
+    return S2N_SUCCESS;
+}
+
+int main(int argc, char **argv)
+{   
+    BEGIN_TEST();
+    
+    /* s2n_send sends NewSessionTicket message and s2n_recv receives it */
+    {
+        struct s2n_connection *server_conn= s2n_connection_new(S2N_SERVER);
+        struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_NOT_NULL(client_conn);
+
+        struct s2n_config *server_config = s2n_config_new();
+        EXPECT_NOT_NULL(server_config);
+
+        struct s2n_config *client_config = s2n_config_new();
+        EXPECT_NOT_NULL(client_config);
+
+        struct s2n_cert_chain_and_key *chain_and_key;
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(server_config, 1));
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, chain_and_key));
+        EXPECT_SUCCESS(s2n_setup_test_ticket_key(server_config));
+
+        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(client_config, 1));
+        EXPECT_SUCCESS(s2n_config_set_session_ticket_cb(client_config, s2n_test_session_ticket_cb, NULL));
+        EXPECT_SUCCESS(s2n_config_disable_x509_verification(client_config));
+
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, server_config));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
+
+        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(server_conn, "default_tls13"));
+        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(client_conn, "default_tls13"));
+
+        DEFER_CLEANUP(struct s2n_stuffer input, s2n_stuffer_free);
+        DEFER_CLEANUP(struct s2n_stuffer output, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, 0));
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&output, 0));
+                                        
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&input, &output, server_conn));
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&output, &input, client_conn));
+
+        EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
+
+        /* Create conditions to send NewSessionTicket message */
+        uint8_t tickets_to_send = 5;
+        server_conn->tickets_to_send = tickets_to_send;
+
+        /* Next message to send will trigger a NewSessionTicket message*/
+        s2n_blocked_status blocked;
+        char message[] = "sent message";
+        EXPECT_SUCCESS(s2n_send(server_conn, message, sizeof(message), &blocked));
+
+        /* Receive NewSessionTicket message */
+        uint8_t data[100];
+        EXPECT_SUCCESS(s2n_recv(client_conn, data, sizeof(data), &blocked));
+
```

Comment:
Add a `EXPECT_BYTEARRAY_EQUAL(data, message, sizeof(message));` to assert that we received the message sent.